server/src/test/scala/org/http4s/server/middleware/GZipSpec.scala

```diff
@@ -20,5 +25,21 @@ class GZipSpec extends Http4sSpec {
       resp.status must_== (Status.Ok)
       resp.headers.get(`Content-Encoding`) must beNone
     }
+
+    checkAll("encoding", new Properties("GZip") {
+      property("middleware encoding == GZIPOutputStream encoding") = forAll { value: String =>
```

Comment:
How about `Vector[Array[Byte]]` here?  Then:

- you can emit each array as a chunk and test the incremental CRC32
- testing binary values instead of just characters (should be the same, but why not?)
- arbitrary binary data will of course be less compressible than arbitrary character data, but I think we don't care